### PR TITLE
Add support of Visual Studio on Windows

### PR DESCRIPTION
**Pull request by generalov**
It uses a slightly improved version of VBScript helper from https://github.com/evgeny-panasyuk/open-in-msvs.el/blob/master/open-in-msvs.vbs for a Visual Studio scripting.

### CODE REVIEW

File: lib/editors/visualstudio/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Windows forced line-endings
+*.vbs text eol=crlf
```
Comment (lahmatiy) [inline lib/editors/visualstudio/.gitattributes]: Does it file really needed?

File: lib/editors/visualstudio/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+var path = require('path');
+
+var helperPath = path.resolve(path.join(__dirname, 'open-in-msvs.vbs'));
```
Comment (lahmatiy) [inline lib/editors/visualstudio/index.js]:
```js
var helperPath = path.resolve(__dirname, 'open-in-msvs.vbs');
```

File: lib/editors/visualstudio/index.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  pattern: '/s /c ""' + helperPath + '" {filename} {line} {column}"'
+};
+
+var detect = function() {
+  var lazy = require('../../detect').lazy('Visual Studio', [], '', {
+    win32: [
+      helperPath
+    ]
+  });
+  return lazy().then(function() { return 'cmd.exe'; });
```
Comment (lahmatiy) [inline lib/editors/visualstudio/index.js]: It's unclear why we need this.
Maybe:
```js
var settings = {
  pattern: '{filename} {line} {column}'
};

var detect = function() {
  if (process.platform === 'win32') {
    return Promise.resolve(helperPath);
  }

  return Promise.reject('"Open in Visual Studio" does not implemented for your platform (' + process.platform + ')')
};
```

File: lib/editors/visualstudio/open-in-msvs.vbs [hunk 4]
```diff
@@ -0,0 +1,80 @@
+' Copyright (c) 2016, Evgeny Panasyuk
```
Comment (lahmatiy) [inline lib/editors/visualstudio/open-in-msvs.vbs]: Is there any chance to avoid VB script?

**Comment by lahmatiy** [comment]
Thank you for the contribution!
I've tried on my PC - it works. I also try my suggestions, works as well.
However, it has a little problem - IDE doesn't become activated on command call. Can we fix it?

**Comment by generalov** [comment]
@lahmatiy Thank you for an advise. I've changed the code as you recommend.

It seems Visual Studio hasn't an sufficient CLI to open a file and go to the line, so some external automation is required. The solution has been inspired by http://stackoverflow.com/questions/350323/open-a-file-in-visual-studio-at-a-specific-line-number

**Comment by lahmatiy** [comment]
Cool!
Could you also relocate files
- `lib/editors/visualstudio/index.js` -> `lib/editors/visualstudio.js`
- `lib/editors/visualstudio/open-in-msvs.vbs` -> `lib/editors/visualstudio.vbs`

Thank you in advance.

**Comment by generalov** [comment]
Done. The the Promise polyfill has added too.

**Comment by lahmatiy** [comment]
Great! Thank you!

**Comment by lahmatiy** [comment]
[2.0.0](https://github.com/lahmatiy/open-in-editor/releases/tag/v2.0.0) is out with VS support 🎉

**Pull request merged**